Enable regional static field and vertical grid processing in the init_atmosphere core

mgduda opened the pull request:
This merge enables the production of static fields and vertical grids for limited-area meshes
in the init_atmosphere core.

mgduda commented:
This PR should probably be merged after #203 , and after adding in checks like [those introduced here](https://github.com/mgduda/MPAS-Model/blob/eb416b71bc61a2d6c8d97a5c719a7dfdd05498f1/src/core_init_atmosphere/mpas_init_atm_static.F#L291-L306) for the MODIS monthly albedo, vegetation fraction, and maximum snow albedo fields added by #203.

MiCurry commented on src/core_init_atmosphere/mpas_init_atm_cases.F:
```diff
@@ -2907,6 +2907,12 @@ subroutine init_atm_case_gfs(block, mesh, nCells, nEdges, nVertLevels, fg, state
             if(ter(iCell) .ne. 0.) then
                do j = 1,nEdgesOnCell(iCell)
+
+                  ! For smoothing at the edge of the mesh, effectively ignore non-existent neighbors
```
I know its getting a bit into the 'what' side of comments, but it might be worth adding the method for ignoring non-existent neighbors i.e. setting them equal to the edge cell.
```
! For smoothing at the edge of the mesh, effectively ignore non-existent neighbors by setting them equal
! to the edge cell
```

mgduda replied:
New comments have been added.

MiCurry commented on src/core_init_atmosphere/mpas_init_atm_static.F:
```diff
@@ -281,8 +292,22 @@ subroutine init_atm_static(mesh, dims, configs)
+             zPixel = sphere_radius * sin(lat_pt)
+             xPixel = sphere_radius * cos(lon_pt) * cos(lat_pt)
+             yPixel = sphere_radius * sin(lon_pt) * cos(lat_pt)
+
```
I think it might be worth having a short comment here (and at further calls to `in_cell`) to explain this section. For instance:

```
! Avoid falsely inflating (or deflating) a regional mesh's border cells by determine if a pixel should be interpolated or ! not
```

mgduda replied:
New comments have been added.

mgduda commented on src/core_init_atmosphere/mpas_init_atm_static.F:
```diff
@@ -827,6 +879,147 @@ real (kind=RKIND) function sphere_distance(lat1, lon1, lat2, lon2, radius)
+!>  Rotates the point (x,y,z) through an angle theta about the vector
+!>  originating at (a, b, c) and having direction (u, v, w).
+!
+!>  Reference: http://inside.mines.edu/~gmurray/ArbitraryAxisRotation/ArbitraryAxisRotation.html
```
I've found what I think is the equivalent page, and I've updated the link.

mgduda commented:
@MiCurry I've rebased the branch onto the latest `develop` so that I can add similar changes to the processing of monthly albedo, monthly vegetation fraction, and maximum snow albedo. I think I've addressed all of your comments with amendments during the rebasing.

mgduda commented on src/core_init_atmosphere/mpas_init_atm_static.F:
```diff
@@ -69,6 +74,7 @@ subroutine init_atm_static(mesh, dims, configs)
  integer, pointer :: supersample_fac
 
  real(kind=RKIND):: lat,lon,x,y
+ real(kind=RKIND):: lat,lon,x,y,z
```
This line should replace the line above. As is, the code fails to compile with ifort. I'm going to correct this with a rebased branch before merging.
